lib/ansible/module_utils/basic.py

```diff
@@ -814,11 +830,15 @@ def __init__(self, argument_spec, bypass_checks=False, no_log=False,
         self._clean = {}
 
         self.aliases = {}
-        self._legal_inputs = ['_ansible_check_mode', '_ansible_no_log', '_ansible_debug', '_ansible_diff', '_ansible_verbosity',
-                              '_ansible_selinux_special_fs', '_ansible_module_name', '_ansible_version', '_ansible_syslog_facility',
-                              '_ansible_socket', '_ansible_shell_executable']
+        self._legal_inputs = ['_ansible_%s' % k for k in PASS_VARS]
         self._options_context = list()
 
+        # set tempdir to remote tmp
+        self.tempdir = os.environ.get('ANSIBLE_REMOTE_TEMP', None)
+
+        # set for all tempfile calls
+        tempfile.tempdir = self.tempdir
```

Comment:
This shouldn't be done.  Modules expect to get the system tempdir by calling tempfile's functions.  Modules are responsible for cleaning up after themselves.  If modules want to delegate that to us, they have the option of passing self.tempdir explicitly to the tempfile functions but we shouldn't be changing the default for them.  This can break other code (even library code) in ways that would be hard to debug.